ui/src/notebooks/context/utils.ts

```diff
@@ -0,0 +1,119 @@
+// Libraries
+import {isEqual, uniq} from 'lodash'
```

Comment:
what would it take to... not?

Reply:
use these? for the uniq, we would have to create a filter function. For the `isEqual`, potentially using a sorted JSON comparison on arrays?